File: week3/c++/week3_4.cpp
using namespace std;

int solution(int num)
{
    int answer = 0;
    long long int n = num;

    while (answer < 500)
    {

        if (n == 1)
        {
            return answer;
        }

        if (n % 2 == 0)
        {
            n /= 2;
        }
        else
        {
            n = n * 3 + 1;
        }
        answer++;
    }

    if (answer == 500)
    {
        answer = -1;
    }
    return answer;
}

File: week3/java/week3_4.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        int cnt = 0;
        long num2 = (long) num;
        while(num2 != 1){
            if(cnt>=500) break;
            if(num2%2==0) num2 /= 2;
            else num2 = num2 * 3 + 1;
            cnt ++;
        }
        if(num2 == 1) answer = cnt;
        else answer = -1;
        return answer;
    }
}